test/vstest.ProgrammerTests/Fakes/FakeAssemblyMetadataProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace vstest.ProgrammerTests.Fakes;
+
+using System.Runtime.Versioning;
+
+using Microsoft.VisualStudio.TestPlatform.CommandLine.Processors;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+internal class FakeAssemblyMetadataProvider : IAssemblyMetadataProvider
+{
+    public FakeFileHelper FakeFileHelper { get; }
+
+    public FakeErrorAggregator FakeErrorAggregator { get; }
+
+    public FakeAssemblyMetadataProvider(FakeFileHelper fakeFileHelper, FakeErrorAggregator fakeErrorAggregator)
+    {
+        FakeFileHelper = fakeFileHelper;
+        FakeErrorAggregator = fakeErrorAggregator;
+    }
+
+    public Architecture GetArchitecture(string filePath)
+    {
+        var file = (FakeTestDllFile)FakeFileHelper.Files.Single(f => f.Path == filePath);
+        return file.Architecture;
+    }
+
+    public FrameworkName GetFrameWork(string filePath)
```

Comment:
I would rename into `GetFramework` to match other namig case.

Reply:
I would as well, but this is the name coming from the interface.